File: src/main/java/com/spirecoder/LongestIncreasingSubsequence.java
package com.spirecoder;

public class LongestIncreasingSubsequence {
    private int getLISLength(int[] arr, int n) {
        int[] lis = new int[n];
        int i, j, max = 0;

        for (i = 0; i < n; i++) {
            lis[i] = 1;
        }

        for (i = 1; i < n; i++) {
            for (j = 0; j < i; j++) {
                if (arr[i] > arr[j] && lis[i] < lis[j] + 1) {
                    lis[i] = lis[j] + 1;
                }
            }
        }

        for (i = 0; i < n; i++) {
            if (max < lis[i]) {
                max = lis[i];
            }
        }
        return max;
    }

    public static void main(String[] args) {
        LongestIncreasingSubsequence lis = new LongestIncreasingSubsequence();
        int[] arr = {50, 3, 10, 7, 40, 80};
        int n = arr.length;
        System.out.println("LIS length: " + lis.getLISLength(arr, n));
    }
}

File: src/main/kotlin/com/spirecoder/LongestIncreasingSubsequence.kt
package com.spirecoder

fun getLISLength(arr: IntArray, n: Int): Int {
    val lis = IntArray(n)
    var i: Int
    var j: Int
    var max = 0

    i = 0
    while (i < n) {
        lis[i] = 1
        i++
    }

    i = 1
    while (i < n) {
        j = 0
        while (j < i) {
            if (arr[i] > arr[j] && lis[i] < lis[j] + 1) {
                lis[i] = lis[j] + 1
            }
            j++
        }
        i++
    }

    i = 0
    while (i < n) {
        if (max < lis[i]) {
            max = lis[i]
        }
        i++
    }
    return max
}


fun main(args: Array<String>) {
    val arr = intArrayOf(50, 3, 10, 7, 40, 80)
    val n = arr.size
    println("LIS length: " + getLISLength(arr, n))
}
